### PR TITLE
TMDM-12612  Show error dialog when switch element between FK and reference.

### PR DESCRIPTION
**Pull request by hbhong**
**What is the current behavior?** (You should also link to an open issue here)
Show error dialog when switch element between FK and reference.
link: https://jira.talendforge.org/browse/TMDM-12612

**What is the new behavior?**
No error dialog and exception are thrown  when switch element between FK and reference.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.mdm.workbench/src/com/amalto/workbench/detailtabs/sections/composites/FKIntegrityComposite.java [hunk 1]
```diff
@@ -109,10 +109,9 @@ protected void checkSubclass() {
 
 	@Override
 	public void dispose() {
+        // Remove listener on disposal (no more need to keep view in sync with model)
```
Comment (lpw82gf) [inline main/plugins/org.talend.mdm.workbench/src/com/amalto/workbench/detailtabs/sections/composites/FKIntegrityComposite.java]: Need to format all the changed code.

Comment (hbhong) [reply main/plugins/org.talend.mdm.workbench/src/com/amalto/workbench/detailtabs/sections/composites/FKIntegrityComposite.java]: Formated it.

File: main/plugins/org.talend.mdm.workbench/src/com/amalto/workbench/detailtabs/sections/composites/FKIntegrityComposite.java [hunk 2]
```diff
@@ -213,27 +212,32 @@ private static void updateModelElement(XSDElementDeclaration xsdComponent,
 		}
 	}
 
-	private class ViewRefreshAdapter implements Adapter {
-		private Notifier notifier;
+    private class ViewRefreshAdapter implements Adapter {
```
Comment (lpw82gf) [inline main/plugins/org.talend.mdm.workbench/src/com/amalto/workbench/detailtabs/sections/composites/FKIntegrityComposite.java]: indentation

File: main/plugins/org.talend.mdm.workbench/src/com/amalto/workbench/detailtabs/sections/composites/FKIntegrityComposite.java [hunk 3]
```diff
@@ -246,16 +250,19 @@ public void setXSDComponent(XSDComponent xsdComponent) {
...
+            removeRefreshAdapter();
+            this.xsdComponent = (XSDElementDeclaration) ((XSDParticle) xsdComponent).getContent();
+            this.xsdComponent.eAdapters().add(refreshAdapter);
+        }
 
 		// Update UI with the information in the XSDComponent
 		refresh();
 	}
+
+    public void removeRefreshAdapter() {
```
Comment (lpw82gf) [inline main/plugins/org.talend.mdm.workbench/src/com/amalto/workbench/detailtabs/sections/composites/FKIntegrityComposite.java]: indentation

**Pull request merged**